python/paddle/fluid/backward.py

```diff
@@ -247,6 +247,116 @@ def _op_can_be_removed_(op_desc, no_grad_set):
     return op_descs
 
 
+def __find_not_need_ops(grad_op_descs, ops, input_grad_names_set):
+    """
+    Pruning Program with Structural Analysis Method of Computational Graph.
+    The nodes of the computational graph composed of backward OPS should be
+    interconnected. If there are unconnected sub-graphs in the computational graph,
+    these sub-graphs should be cut off.
```

Comment:
Maybe better to add comments for grad_op_descs, ops, input_grad_names_set

Reply:
Done